### PR TITLE
[SPARK-24601][SPARK-27051][BACKPORT][CORE] Update to Jackson 2.9.8

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

This backports:
https://github.com/apache/spark/commit/ab1650d2938db4901b8c28df945d6a0691a19d31
https://github.com/apache/spark/commit/7857c6d633f3df426a6ac4618316eb83b1cefe2b

which collectively updates Jackson to 2.9.8.

## How was this patch tested?

Existing tests.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -224,6 +224,10 @@
       <groupId>org.scala-lang</groupId>
       <artifactId>scala-library</artifactId>
     </dependency>
+    <dependency>
```
Comment (srowen) [inline core/pom.xml]: We used to get this transitively from Jackson; need to make it explicit now.

File: core/src/main/scala/org/apache/spark/deploy/rest/SubmitRestProtocolMessage.scala [hunk 2]
```diff
@@ -36,7 +36,7 @@ import org.apache.spark.util.Utils
  *   (2) the Spark version of the client / server
  *   (3) an optional message
  */
-@JsonInclude(Include.NON_NULL)
+@JsonInclude(Include.NON_ABSENT)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/deploy/rest/SubmitRestProtocolMessage.scala]: This is the core of the change to maintain behavior, as NON_NULL changed behavior in 2.7 and 2.8

**Comment by dongjoon-hyun** [comment]
cc @dbtsai

**Comment by arunmahadevan** [comment]
LGTM

**Comment by dbtsai** [comment]
LGTM.

**Review by felixcheung — APPROVED** [review]

File: core/pom.xml (hunk 1 above)
Comment (felixcheung) [reply core/pom.xml]: ah, this one is nasty, we've had soo many conflict on this one..

Comment (srowen) [reply core/pom.xml]: This much isn't a change to the final dependency graph, at least. It always need/depended on scala-reflect

**Review by Fokko — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to branch-2.4.

This is a collective backport (except 3.0-only `ExecutorMetricsJsonDeserializer` class).

Thank you, @srowen , @dbtsai , @felixcheung , @arunmahadevan , @Fokko !

**Comment by srowen** [comment]
I support this, and wanted to more fully explain the logic, as it occurs to me not all of it was visible in the conversation here:

- The Jackson security issue has come up a few times on JIRAs
- It was actually raised again on the security@ list for the project last week
- I can also say this has been raised by a few customers at Databricks, FWIW
- Having the older version is also beginning to cause problems with apps that want to include third-party libs that depend on newer Jackson
- There's a perf boost

Weighing that against the relatively small behavior change I think it's the right thing for 2.4.x.

This won't go into 2.4.2 unless there is another RC. If not, it'll be in 2.4.3.

**Pull request closed without merging**